src/GitVersionCore.Tests/TestFileSystem.cs

```diff
@@ -1,52 +1,188 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Text;
+
 using GitVersion.Helpers;
 
+using LibGit2Sharp;
+
+using NSubstitute;
+
 public class TestFileSystem : IFileSystem
 {
     Dictionary<string, string> fileSystem = new Dictionary<string, string>();
 
+
     public void Copy(string @from, string to, bool overwrite)
     {
         throw new NotImplementedException();
     }
 
+
     public void Move(string @from, string to)
     {
         throw new NotImplementedException();
     }
 
+
     public bool Exists(string file)
     {
-        return fileSystem.ContainsKey(file);
+        return this.fileSystem.ContainsKey(file);
     }
 
+
     public void Delete(string path)
     {
         throw new NotImplementedException();
     }
 
+
     public string ReadAllText(string path)
     {
-        return fileSystem[path];
+        return this.fileSystem[path];
     }
 
+
     public void WriteAllText(string file, string fileContents)
     {
-        if (fileSystem.ContainsKey(file))
-            fileSystem[file] = fileContents;
+        if (this.fileSystem.ContainsKey(file))
+        {
+            this.fileSystem[file] = fileContents;
+        }
         else
-            fileSystem.Add(file, fileContents);
+        {
+            this.fileSystem.Add(file, fileContents);
+        }
     }
 
+
     public IEnumerable<string> DirectoryGetFiles(string directory, string searchPattern, SearchOption searchOption)
     {
         throw new NotImplementedException();
     }
 
+
     public Stream OpenWrite(string path)
     {
         return new TestStream(path, this);
     }
+
+
+    public Stream OpenRead(string path)
+    {
+        if (this.fileSystem.ContainsKey(path))
+        {
+            var content = this.fileSystem[path];
+            return new MemoryStream(Encoding.UTF8.GetBytes(content));
+        }
+
+        throw new FileNotFoundException("File not found.", path);
+    }
+
+
+    public void CreateDirectory(string path)
+    {
+    }
+
+
+    public long GetLastDirectoryWrite(string path)
+    {
+        return 1;
+    }
+
+
+    public string TreeWalkForDotGitDir(string directory)
+    {
+        return directory;
+    }
+
+
+    public IRepository GetRepository(string gitDirectory)
```

Comment:
Why not use the Fixture stuff? Mocking out libgit2sharp stuff is annoying in the long run I think